drivers/ethernet/eth_mcux.c

```diff
@@ -519,18 +520,24 @@ static void eth_mcux_phy_event(struct eth_context *context)
 		/* PHY Basic status is available. */
 #if defined(CONFIG_ETH_MCUX_NO_PHY_SMI) && ETH_MCUX_FIXED_LINK
 		link_up = true;
+		autoneg_complete = true;
 #else
 		status = ENET_ReadSMIData(context->base);
 		link_up =  status & PHY_BSTATUS_LINKSTATUS_MASK;
+		autoneg_complete = status & PHY_BSTATUS_AUTONEGCOMP_MASK;
```

Comment:
why we add the autoneg complete check? per my understanding when link is up the autoneg shall finished. see
https://www.ieee802.org/3/bp/public/nov14/mcclellan_3bp_03_1114_%20Autoneg_baseline_text_proposal_v0p4.pdf page 28. Figure 888-14 - Arbitration state diagram. what special issue you find if without this check?

Reply:
You're right, I wasn't aware that the auto negotiation would always be complete when the link was marked up. I added the check during testing, and kept it in place because the SDK examples check both flags. I've removed the auto negotiation check in the latest push, and everything still seems to work fine